config/nrfconnect/chip-module/Kconfig

```diff
@@ -239,9 +267,9 @@ config CHIP_DEVICE_GENERATE_ROTATING_DEVICE_UID
 endif # CHIP_FACTORY_DATA_BUILD
 
 # See config/zephyr/Kconfig for full definition
-config CHIP_FACTORY_RESET_ERASE_NVS
-	bool
+config CHIP_FACTORY_RESET_ERASE_SETTINGS
 	default y
+	depends on NVS || ZMS
```

Comment:
The definition in `config/zephyr/Kconfig` already contains `depends on SETTINGS_NVS || SETTINGS_ZMS`. Isn't it enough?

Reply:
If I get it correctly, if we use default to `y` here we overwrite the whole `CHIP_FACTORY_RESET_ERASE_SETTINGS` from zephyr's definition, so `depends on SETTINGS_NVS || SETTINGS_ZMS` from zephyr has no power here. @Damian-Nordic am I right?

Reply:
ah, yes, you're right. Nevermind then :)